### DISCUSSION TITLE
How can I set email&password? -> YoutubeAccessTokenTracker

### DISCUSSION THREAD

**Discussion by Kodeekk** [lavalink-devs/lavaplayer#66] [Q&A] [answered]
`[11:55:18] [lava-daemon-pool-info-loader-2-thread-1/WARN] (YoutubeAccessTokenTracker) YouTube auth tokens can't be retrieved because email and password is not set in YoutubeAudioSourceManager, age restricted videos will throw exceptions.
[11:55:19] [lava-daemon-pool-info-loader-2-thread-1/INFO] (YoutubeAccessTokenTracker) Updating YouTube visitor id (current is null).`

**Comment by topi314** [comment, marked as answer]
https://github.com/lavalink-devs/lavaplayer/blob/cb2b849862116477adade854f87e8bfc12d7d45e/main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/YoutubeAudioSourceManager.java#L71

**Reply by Kodeekk** [reply]
in source code of lavaplayer it looks like this:
`public YoutubeAudioSourceManager(
      boolean allowSearch,
      YoutubeTrackDetailsLoader trackDetailsLoader,
      YoutubeSearchResultLoader searchResultLoader,
      YoutubeSearchMusicResultLoader searchMusicResultLoader,
      YoutubeSignatureResolver signatureResolver,
      YoutubePlaylistLoader playlistLoader,
      YoutubeLinkRouter linkRouter,
      YoutubeMixLoader mixLoader
  )`
there`s no email or password args :(

**Reply by topi314** [reply]
you just need to do
```java
new YoutubeAudioSourceManager(allowSearch, email, password);
```

you can ignore all the other constructors

**Reply by Kodeekk** [reply]
I barely know Java at all (I use Kotlin) and have no idea how to apply everything you wrote to my code. Sorry, I'm too dumb

**Reply by Kodeekk** [reply]
I'm just trying to write 5 functions like play, pause, resume, stop and volume, but here I get this error(

**Reply by topi314** [reply]
in kotlin you would simply omit the `new` keyword then
```kt
val audioPlayerManager = DefaultAudioPlayerManager()
audioPlayerManager.registerSourceManager(YoutubeAudioSourceManager(allowSearch, email, password))
```
however if you previously called [`AudioSourceManagers.registerRemoteSources`](https://github.com/lavalink-devs/lavaplayer/blob/cb2b849862116477adade854f87e8bfc12d7d45e/main/src/main/java/com/sedmelluq/discord/lavaplayer/source/AudioSourceManagers.java#L24)

you now need to add all source managers on your own, else you would register yt twice
